### PR TITLE
Add v2 base code

### PR DESCRIPTION
**Pull request by Mecharyry**
### Problem
Moving towards version two of download manager.

### Solution
Add the code that @zegnus has kindly provided us from https://github.com/zegnus/lite-download-manager 

The removal of download-manager v1 was done on the epic branch, just a delete all from the `download-manager` package 😂 

**Note** this does not include any unit tests, this is simply replacing the code.

### Screen Capture
download

### CODE REVIEW

**Comment by Electryc** [comment]
>Remove the old code and replace

`+4,903 −0`

I don't see remove 😛

**Comment by Mecharyry** [comment]
😂  I removed the code from the epic branch, I thought it would be super not helpful to have it all in here in the end @Electryc

File: team-props/lint-rules/config.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<lint>
+
+  <issue id="all">
```
Comment (Mecharyry) [inline team-props/lint-rules/config.xml]: Had to ignore all lint warnings from the generated package, Room seems to have more than a few issues. We can decide to address these later if we decide.

**Comment by Electryc** [comment]
@Mecharyry I see 👍

**Pull request merged**